Source/engine.h

```diff
@@ -598,7 +605,40 @@ void SetRndSeed(int32_t s);
 int32_t AdvanceRndSeed();
 int32_t GetRndSeed();
 int32_t GenerateRnd(int32_t v);
-std::unique_ptr<BYTE[]> LoadFileInMem(const char *pszName, DWORD *pdwFileLen = nullptr);
+
+size_t GetFileSize(const char *pszName);
+void LoadFileData(const char *pszName, byte *buffer, size_t bufferSize);
+
+template <typename T, std::size_t N>
+void LoadFileInMem(const char *path, std::array<T, N> &data)
+{
+	LoadFileData(path, reinterpret_cast<byte *>(&data), N * sizeof(T));
+}
+
+/**
+ * @brief Load a file in to a buffer
+ * @param path Path of file
+ * @param pdwFileLen Number of

```

Comment:
Guess this should be something like
```suggestion
 * @param elements Number of T elements read
```